climada/engine/impact.py

```diff
@@ -84,164 +487,157 @@ class Impact():
         only filled if save_mat is True in calc()
     """
 
-    def __init__(self):
-        """Empty initialization."""
-        self.tag = dict()
-        self.event_id = np.array([], int)
-        self.event_name = list()
-        self.date = np.array([], int)
-        self.coord_exp = np.ndarray([], float)
-        self.crs = DEF_CRS
-        self.eai_exp = np.array([])
-        self.at_event = np.array([])
-        self.frequency = np.array([])
-        self.tot_value = 0
-        self.aai_agg = 0
-        self.unit = ''
-        self.imp_mat = sparse.csr_matrix(np.empty((0, 0)))
+    def __init__(self,
+                 event_id=None,
+                 event_name=None,
+                 date=None,
+                 frequency=None,
+                 coord_exp=None,
+                 crs=DEF_CRS,
+                 eai_exp=None,
+                 at_event=None,
+                 tot_value=0,
+                 aai_agg=0,
+                 unit='',
+                 imp_mat=None,
+                 tag=None):
+
+        self.tag = tag or {}
+        self.event_id = np.array([], int) if event_id is None else event_id
+        self.event_name = event_name or []
+        self.date = np.array([], int) if date is None else date
+        self.coord_exp = np.ndarray([], float) if coord_exp is None else coord_exp
+        self.crs = crs
+        self.eai_exp = np.array([], float) if eai_exp is None else eai_exp
+        self.at_event = np.array([], float) if at_event is None else at_event
+        self.frequency = np.array([],float) if frequency is None else frequency
+        self.tot_value = tot_value
+        self.aai_agg = aai_agg
+        self.unit = unit
+        self.imp_mat = sparse.csr_matrix(np.empty((0, 0))) if imp_mat is None else imp_mat
 
-    def calc_freq_curve(self, return_per=None):
-        """Compute impact exceedance frequency curve.
+    def calc(self, exposures, impact_funcs, hazard, save_mat=False):
+        """This function is deprecated, use Impact.calc_risk and Impact.calc_insured_risk instead.
+        """
+        impcalc= ImpactCalc(exposures, impact_funcs, hazard)
+        if ('deductible' in exposures.gdf) and ('cover' in exposures.gdf) \
+            and exposures.gdf.cover.max():
+            # LOGGER.warning("To compute the risk transfer value"
+            #      "please use Impact.calc_insured_risk")
+            self.__dict__ = impcalc.insured_impact(save_mat).__dict__
+        else:
+            # LOGGER.warning("The use of Impact.calc() is deprecated."
+            #      "Please use Impact.calc_risk() or Impact.calc_risk_insured().")
+            self.__dict__ = impcalc.impact(save_mat).__dict__
+
+#TODO: new name
+    @classmethod
+    def from_eih(cls, exposures, impfset, hazard,
```

Comment:
There were already several suggestions, so I'll just add another one for the name of this method:
`from_exp_impf_haz` 😅

Reply:
This is a bit long for my taste :D But I have no strong opinion here either ^^.